### PR TITLE
Cleaning quantum-tensor connection code

### PR DESCRIPTION
**Pull request by sneakyweasel**
The game is working smoothly with a very nice performance increase, this PR is mostly to discuss a few things:

- Grid information is sent to qt in JSON format and there is a mismatch between the JSON format we are using in QT and QG2, we should discuss what the best data format could be and stick to it ;p
- References to qt operators, transitions, etc. from the Vue front-end have been removed and it feels much nicer to have a single point of comm. Still need to fix EncyclopediaBoard.
- EncyclopediaBoard and Board are very similar and it might be a good idea to have only one of them to avoid duplication.
- QuantumFrame and QuantumSimulation were moved to QT and this version only uses QT Simulation as a input/output point bypassing the need for Photons since there's a lot of duplicate code.

Waiting for your comments!

### CODE REVIEW

File: src/components/AppButton.vue [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { Vue, Component, Prop } from 'vue-property-decorator'
 
 @Component
 export default class AppButton extends Vue {
-  @Prop() readonly hoverColor!: string
+  // @Prop() readonly hoverColor!: string
```
Comment (stared) [inline src/components/AppButton.vue:12]: How is it related to performance? 
AND
If anything, we delete things, not comment them.

File: src/components/EncyclopediaPage/EncyclopediaBoard.vue [hunk 2]
```diff
@@ -121,24 +118,31 @@ export default class EncyclopediaBoard extends Vue {
...
+    this.simulation = new Simulation(this.grid.exportSimGrid())
     if (this.initialState.length === 1) {
       const d = this.initialState[0]
-      this.simulation.intializeFromXYState(d.posX, d.posY, d.vecDirPol)
+      // this.simulation.intializeFromXYState(d.posX, d.posY, d.vecDirPol)
+      this.simulation.initializeFromIndicator({
+        x: d.posX,
+        y: d.posY,
+        rotation: 'v',
+        polarization: 'v',
```
Comment (stared) [inline src/components/EncyclopediaPage/EncyclopediaBoard.vue:129]: I guess capital `'V'`

File: src/components/GamePage/index.vue [hunk 3]
```diff
@@ -112,6 +111,7 @@ import AppButton from '@/components/AppButton.vue'
 import AppOverlay from '@/components/AppOverlay.vue'
 import { getRockTalkIdByLevelId } from '@/components/RockTalkPage/loadRockTalks'
 import type { ActionMethod } from 'vuex';
+import { IAbsorption } from 'quantum-tensors/dist/interfaces'
```
Comment (stared) [inline src/components/GamePage/index.vue:114]: Not sure if it is a correct import.

File: src/components/GamePage/index.vue [hunk 4]
```diff
@@ -344,22 +358,30 @@ export default class Game extends Vue {
...
-  get pathParticles(): Particle[] {
-    return uniq(this.simulation.allParticles)
+  get pathParticles(): IParticle[] {
+    const result: IParticle[] = []
+    this.simulation.frames.forEach((frame): void => {
+      frame.particles.forEach((particle): void => {
+        result.push(particle)
+      })
+    })
+    return uniq(result)
```
Comment (stared) [inline src/components/GamePage/index.vue:368]: Never use `forEach` / `push` when you can use `map` (or `flatMap`) instead.

File: src/components/GamePage/index.vue [hunk 5]
```diff
@@ -344,22 +358,30 @@ export default class Game extends Vue {
...
     return this.simulation.frames[this.frameIndex]
   }
 
   /**
    * Current simulation frame particles
    */
   get particles(): Particle[] {
-    return this.activeFrame.particles
+    return this.activeFrame.particles.map((iparticle) => {
+      return Particle.importParticle(iparticle)
```
Comment (stared) [inline src/components/GamePage/index.vue:383]: Remove `{` and `return`.

File: src/engine/Element.ts [hunk 6]
```diff
@@ -1,5 +1,4 @@
-import * as qt from 'quantum-tensors'
-import { Elem, Group, ITransition } from './interfaces'
+import { Elem, Group } from './interfaces'
```
Comment (stared) [inline src/engine/Element.ts:1]: i would go for `@/engine/interfaces` for consistency.

File: src/components/AppButton.vue (hunk 1 above)
Comment (sneakyweasel) [reply src/components/AppButton.vue:12]: Will delete it, it left a linter error.

File: src/components/EncyclopediaPage/EncyclopediaBoard.vue (hunk 2 above)
Comment (sneakyweasel) [reply src/components/EncyclopediaPage/EncyclopediaBoard.vue:129]: I'm having trouble to understand how the indicator differs from the "normal" Board here, do you think we could use only one board or is there too much specific code?

**Review by stared — CHANGES_REQUESTED** [review]
Separation going in the right way.

Some things (e.g. `push` should be the last resort, we already discussed that).

The current code has errors in CI.

**Comment by stared** [comment]
Things like unifying Boards for game and encyclopedia - clearly another PR (with which I would like to wait for @Frizi input - as a lot of things on board, especially related to interaction, may need serious refactoring).

File: src/components/GamePage/index.vue (hunk 3 above)
Comment (sneakyweasel) [reply src/components/GamePage/index.vue:114]: Right now there's a mismatch between IAbsorption in QT and QG2, one using Coord and the other x, y. Will write a post on Basecamp about those issues.

**Comment by sneakyweasel** [comment]
Everything seems to be working fine expect the indicator for bra-ket-view which somehow breaks and gives a strange error whose only reference is https://github.com/microsoft/TypeScript/issues/28102

**Comment by sneakyweasel** [comment]
Everything seems to be working fine. CI tests are passing.

**Review by Frizi — CHANGES_REQUESTED** [review]

**Comment by Frizi** [comment]
> Things like unifying Boards for game and encyclopedia - clearly another PR (with which I would like to wait for @Frizi input - as a lot of things on board, especially related to interaction, may need serious refactoring).

I'm all for it :) As you said, I wouldn't hold this PR back in wait for that change, but I'd definitely like to get rid of redundancy here.

File: src/components/GamePage/GameGraph.vue [hunk 7]
```diff
@@ -1,62 +1,17 @@
...
-
       <!-- NODE -->
       <g :style="computeNodeStyle">
         <g v-for="(node, i) in nodes" :key="'node' + i" :class="computeNodeClass(node)">
-          <circle :cx="node.x" :cy="node.y" r="5" @mouseover="handleMouseOver(node.fIndex)" />
-          <!-- <text class="nodeText" :x="node.x" :y="node.y + 4" text-anchor="middle">
-            {{ node.label }}
-          </text> -->
+          <circle :cx="node.x" :cy="node.y" r="5" @click="handleClick(node.fIndex)" />
+          <!-- <text class="nodeText" :x="node.x" :y="node.y + 4" text-anchor="middle">{{ node.label }}</text> -->
```
Comment (Frizi) [inline src/components/GamePage/GameGraph.vue:8]: unnecessary commented code

**Comment by stared** [comment]
I tried to test it locally, but there is no visible speedup. 

(Plus, it did add a multiverse graph. @sneakyweasel, was it an intended change?)

**Comment by stared** [comment]
There are some conflicts, could you rebase (or at least merge fix conflicts, then I will squash-rebase this part).

Screenshot 2020-09-07 at 19 47 10

**Review by stared — APPROVED** [review]

**Pull request merged**